rtree/rtree.go

```diff
@@ -14,11 +14,9 @@ type node struct {
 
 // entry is an entry under a node, leading either to terminal items, or more nodes.
 type entry struct {
-	box Box
-
-	// For leaf nodes, this is a record ID. For non-leaf nodes, this is the
-	// index of the child node.
-	index int
+	box      Box
+	child    int    // Only populated for non-leaf nodes.
+	recordID uint64 // Only populated for leaf nodes.
```

Comment:
`child` and `recordID` could _technically_ be stored in the same variable. However, I'm going to come back and revisit the internal structure of the rtree anyway (there are a few low level efficiency gains to be had, e.g. using arrays rather than slices).